src/frontend/apps/standalone_site/src/features/Language/hook/useLanguage.spec.tsx

```diff
@@ -0,0 +1,44 @@
+import { renderHook, waitFor } from '@testing-library/react';
+
+import { useLanguageStore } from '../store/languageStore';
+
+import { useLanguage } from './useLanguage';
+
+const consoleWarn = jest
+  .spyOn(console, 'warn')
+  .mockImplementation(() => jest.fn());
+
+describe('hook/useLanguage', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+    consoleWarn.mockClear();
+  });
+
+  it('checks intl is correctly init', async () => {
+    const { result } = renderHook(() => useLanguage());
+
+    await waitFor(() => {
+      expect(result.current).toBeDefined();
+    });
+
+    expect(result.current?.locale).toEqual('en-US');
+    expect(result.current?.defaultLocale).toEqual('en');
+    expect(consoleWarn).toHaveBeenCalled();
```

Comment:
What is the point in testing that console warning has been called ?

Reply:
The warning is a wanted behavior, so we check that the call is made.
https://github.com/openfun/marsha/blob/feature/anthony/language-switcher/src/frontend/apps/standalone_site/src/features/Language/utils.ts#L70

It could be more accurate though:
```suggestion
    expect(consoleWarn).toHaveBeenCalledWith('[intl] No translation found for language en-US');
```

Reply:
[fixup! ✨(website) create language feature](https://github.com/openfun/marsha/pull/2366/commits/cb07b4b507b9b2a5a7c403f5d40a77b59aaa663b)